ps_facetedsearch.php

```diff
@@ -741,468 +340,523 @@ public static function fullPricesIndexProcess($cursor = 0, $ajax = false, $smart
-                        $min_price[$currency['id_currency']] = $price;
+                }
+
+                foreach ($groups as $group) {
+                    foreach ($currencyList as $currency) {
+                        $price = Product::priceCalculation(
+                            $idShop,
+                            (int) $idProduct,
+                            (int) $idCountry,
+                            null,
+                            null,
+                            null,
+                            (int) $currency['id_currency'],
+                            (int) $group['id_group'],
+                            null,
+                            false,
+                            6,
+                            false,
+                            true,
+                            true,
+                            $specificPriceOutput,
+                            true
+                        );
+
+                        if (!isset($maxPrice[$idCountry][$currency['id_currency']])) {
+                            $maxPrice[$idCountry][$currency['id_currency']] = 0;
+                        }
+
+                        if (!isset($minPrice[$idCountry][$currency['id_currency']])) {
+                            $minPrice[$idCountry][$currency['id_currency']] = null;
+                        }
+
+                        if ($price > $maxPrice[$idCountry][$currency['id_currency']]) {
+                            $maxPrice[$idCountry][$currency['id_currency']] = $price;
+                        }
+
+                        if ($price == 0) {
+                            continue;
+                        }
+
+                        if (null === $minPrice[$idCountry][$currency['id_currency']] || $price < $minPrice[$idCountry][$currency['id_currency']]) {
+                            $minPrice[$idCountry][$currency['id_currency']] = $price;
+                        }
                     }
                 }
             }
 
-            $values = array();
-            foreach ($currency_list as $currency) {
-                $values[] = '('.(int) $id_product.',
-					'.(int) $currency['id_currency'].',
-					'.$id_shop.',
-					'.(int) $min_price[$currency['id_currency']].',
-					'.(int) Tools::ps_round($max_price[$currency['id_currency']] * (100 + $max_tax_rate) / 100, 0).')';
+            $values = [];
+            foreach ($taxRatesByCountry as $taxRateByCountry) {
+                $taxRate = $taxRateByCountry['rate'];
+                $idCountry = $taxRateByCountry['id_country'];
+                foreach ($currencyList as $currency) {
+                    $minPriceValue = array_key_exists($idCountry, $minPrice) ? $minPrice[$idCountry][$currency['id_currency']] : 0;
+                    $maxPriceValue = array_key_exists($idCountry, $maxPrice) ? $maxPrice[$idCountry][$currency['id_currency']] : 0;
+                    if ($taxRateByCountry['iso_code'] !== self::ISO_CODE_US) {
```

Comment:
I have a question about this special feature, is US the ONLY country which needs this? Maybe it should be an array like `taxFreeCountries` or something similar?

Reply:
Nope, it's the only one :/